### PR TITLE
Add SSL support

### LINKED ISSUE #44: Enable HTTPS protocol

### PR DESCRIPTION
**Pull request by ffdarkpenguin**
closes #44

---
This change is [Reviewable](https://reviewable.io/reviews/hewlettpackard/oneview-redfish-toolkit/91)

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -14,7 +14,49 @@
...
 from oneview_redfish_toolkit.app import app
 
 if __name__ == "__main__":
-    app.run(host="0.0.0.0", port=5000, debug=True)
+    config = configparser.ConfigParser()
+    config.optionxform = str
+    try:
+        config.read('redfish.conf')
+    except Exception:
+        print("Failed to load config file. Aborting...")
```
Comment (llvieira) [inline run.py]: I would use Python logging here... 'Print' is too bad :)

File: run.py [hunk 2]
```diff
@@ -14,7 +14,49 @@
...
+        exit(1)
+
+    try:
+        port = int(config["redfish"]["redfish_port"])
+    except Exception:
+        print("Port must be an integer number between 1 and 65536")
```
Comment (llvieira) [inline run.py]: Python logging...

File: run.py [hunk 3]
```diff
@@ -14,7 +14,49 @@
...
+        exit(1)
+    # Checking port range
+    if port < 1 or port > 65536:
+        print("Port must be an integer number between 1 and 65536")
```
Comment (llvieira) [inline run.py]: Python logging...

File: run.py [hunk 4]
```diff
@@ -14,7 +14,49 @@
...
+        exit(1)
+
+    ssl_type = config["ssl"]["SSLType"]
+    # Check SSLType:
+    if ssl_type not in ('disabled', 'adhoc', 'certs'):
+        print("Invalid SSL type: {}. Must be one of: None, adhoc or certs".
```
Comment (llvieira) [inline run.py]: Python logging...

File: run.py [hunk 5]
```diff
@@ -14,7 +14,49 @@
...
+    if ssl_type == 'disabled':
+        app.run(host="0.0.0.0", port=5000, debug=True)
+    elif ssl_type == 'adhoc':
+        app.run(host="0.0.0.0", port=5000, debug=True, ssl_context="adhoc")
+    else:
+        # We should use certs file provided by the user
+        ssl_cert_file = config["ssl"]["SSLCertFile"]
+        ssl_key_file = config["ssl"]["SSLKeyFile"]
+        if ssl_cert_file == "" or ssl_key_file == "":
+            print("SSL type: is 'cert' but one of the files are missing on"
```
Comment (llvieira) [inline run.py]: Python logging...

**Comment by llvieira** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/91)*

**Comment by ffdarkpenguin** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[run.py, line 28 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/91#-Kv8zp0st9m6AAs-n6EK-r1-28:-Kv9SdOGNxt8IrtDCoUg:b-si0ysz) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/db82ad3a1f75cae1edf2b08bb78dd451972bd003/run.py#L28)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

I would use Python logging here... 'Print' is too bad :)
</blockquote></details>

Only if we are to change the config load from app to the run.py. It makes no sense IMO to rewrite de log config only to use in here. Wait for po decision on that

---

*[run.py, line 34 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/91#-Kv8zp0st9m6AAs-n6EK-r1-34:-Kv9T9XHgqW4__xX84A0:bt5vefq) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/db82ad3a1f75cae1edf2b08bb78dd451972bd003/run.py#L34)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

Python logging...
</blockquote></details>

see replay above

---

*[run.py, line 38 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/91#-Kv8zp0st9m6AAs-n6EK-r1-38:-Kv9TBfdgIuLiCyB_ETx:bb7od0r) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/db82ad3a1f75cae1edf2b08bb78dd451972bd003/run.py#L38)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

Python logging...
</blockquote></details>

same

---

*[run.py, line 44 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/91#-Kv8zp0st9m6AAs-n6EK-r1-44:-Kv9TCXYP3ccdSeOHccj:bb7od0r) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/db82ad3a1f75cae1edf2b08bb78dd451972bd003/run.py#L44)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

Python logging...
</blockquote></details>

same

---

*[run.py, line 57 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/91#-Kv8zp0st9m6AAs-n6EK-r1-57:-Kv9TDKjwnUC6W7_xiyu:bb7od0r) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/db82ad3a1f75cae1edf2b08bb78dd451972bd003/run.py#L57)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

Python logging...
</blockquote></details>

same

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/91)*

**Comment by bsoutham** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/91#-:-KvEKzzZST-y78-F9Nk3:bnfp4nl)*

**Pull request merged**